src/content/blog/astro-4150.mdx

```diff
@@ -0,0 +1,184 @@
+
+```astro
+---
+// src/pages/newsletter.astro
+import { actions } from 'astro:actions';
+
+const result = Astro.getActionResult(actions.newsletter);
+---
+
+{result && !result.error && <p>Thanks for signing up!</p>}
+
+<form method="POST" action={actions.newsletter}>
+  <input type="email" name="email" />
+  <button>Sign up</button>
+</form>
+```
+
+For more information and usage examples, see our [all-new Actions guide](https://docs.astro.build/en/guides/actions).
+
+## Rewrites in i18n fallbacks
+
+A new option was added to [internationalization (i18n) routing](https://docs.astro.build/en/guides/internationalization/) which allows more control over how fallbacks are handled. You can now specify whether a fallback results in a rewrite (a `200` response from the server) or a redirect, the current default.
+
+The fallback system in i18n allows you to define which content is used for a locale when that content is not available in the user's locale. For example, your config might look like this:
+
+```js
+import { defineConfig } from "astro/config"
+export default defineConfig({
+  i18n: {
+    defaultLocale: "en",
+    locales: ["es", "en", "fr"],
+    fallback: {
+      fr: "es"
+    },
+    routing: {
+      fallbackType: "rewrite"
+    }
+  }
+})
+```
+
+For more information on this new feature, see [our revamped documentation on handling fallbacks in i18n](https://docs.astro.build/en/guides/internationalization/#fallback).
+
+## libSQL remotes in Astro DB
+
+You can now specify a [libSQL](https://turso.tech/libsql) server instance as the remote for `@astrojs/db`. This allows you to self-host your own libSQL server as an alternative to using Astro DB with Astro Studio. This option works with any supported libSQL protocol.
+
+To use this feature, set the following environment variables:
+
+- `ASTRO_DB_REMOTE_URL`: the connection URL to your libSQL server.
+- `ASTRO_DB_APP_TOKEN`: the auth token to your libSQL server.
+
+Enabling this feature allows you to connect to the libSQL server in your production environment and also allows you to use the Astro DB deployment and push (migration) commands with the libSQL instance. Thanks to [Fryuni](https://github.com/Fryuni) for this amazing contribution!
+
+For more information on using libSQL servers in Astro DB, [visit our Astro DB guide](https://docs.astro.build/en/guides/astro-db/#libsql).
+
+## Timeout option in `client:idle`
+
+The `client:idle` directive is used to load an [Astro Island](https://docs.astro.build/en/concepts/islands/) when the user's CPU is idle (which no other JavaScript is executing). Under the hood, this feature relies on the [requestIdleCallback](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback) feature of browsers.
+
+In Astro 4.15, you can now specify a `timeout` property that will trigger the island to load if it has not already done so in that time. This is useful to add priority to a component that might be competing with other idle components.
```

Comment:
```suggestion
In Astro 4.15, you can now specify a `timeout` property that will trigger hydration to ensure your element is interactive within a specified time frame. This is useful to add priority to a component that might be competing with other idle components.
```
We'd said "load" a lot, but not yet "hydration", so this is a bit of word variety.